packages/lsp-extension/src/index.ts

```diff
@@ -0,0 +1,265 @@
+import { ITranslator } from '@jupyterlab/translation';
+import {
+  IFormComponentRegistry,
+  LabIcon,
+  pythonIcon
+} from '@jupyterlab/ui-components';
+import { PartialJSONObject } from '@lumino/coreutils';
+import { Signal } from '@lumino/signaling';
+
+import { renderServerSetting } from './renderer';
+
+import type { FieldProps } from '@rjsf/core';
+const plugin: JupyterFrontEndPlugin<ILSPDocumentConnectionManager> = {
+  activate,
+  id: '@jupyterlab/lsp-extension:plugin',
+  requires: [ISettingRegistry, ITranslator],
+  optional: [IRunningSessionManagers, IFormComponentRegistry],
+  provides: ILSPDocumentConnectionManager,
+  autoStart: true
+};
+
+const featurePlugin: JupyterFrontEndPlugin<ILSPFeatureManager> = {
+  id: '@jupyterlab/lsp-extension:feature',
+  activate: () => new FeatureManager(),
+  provides: ILSPFeatureManager,
+  autoStart: true
+};
+
+const codeExtractorManagerPlugin: JupyterFrontEndPlugin<ILSPCodeExtractorsManager> =
+  {
+    id: ILSPCodeExtractorsManager.name,
+    activate: app => {
+      const extractorManager = new CodeExtractorsManager();
+
+      const markdownCellExtractor = new TextForeignCodeExtractor({
+        language: 'markdown',
+        isStandalone: false,
+        file_extension: 'md',
+        cellType: ['markdown']
+      });
+      extractorManager.register(markdownCellExtractor, null);
+      const rawCellExtractor = new TextForeignCodeExtractor({
+        language: 'text',
+        isStandalone: false,
+        file_extension: 'txt',
+        cellType: ['raw']
+      });
+      extractorManager.register(rawCellExtractor, null);
+      return extractorManager;
+    },
+    provides: ILSPCodeExtractorsManager,
+    autoStart: true
+  };
+
+/**
+ * Activate the lsp plugin.
+ */
+function activate(
+  app: JupyterFrontEnd,
+  settingRegistry: ISettingRegistry,
+  translator: ITranslator,
+  runningSessionManagers: IRunningSessionManagers | null,
+  settingRendererRegistry: IFormComponentRegistry | null
+): ILSPDocumentConnectionManager {
+  const LANGUAGE_SERVERS = 'languageServers';
+  const languageServerManager = new LanguageServerManager({});
+  const connectionManager = new DocumentConnectionManager({
+    languageServerManager
+  });
+
+  const updateOptions = (settings: ISettingRegistry.ISettings) => {
+    const options = settings.composite as Required<LanguageServers>;
+
+    const languageServerSettings = (options.languageServers ||
+      {}) as TLanguageServerConfigurations;
+
+    connectionManager.initialConfigurations = languageServerSettings;
+    // TODO: if priorities changed reset connections
+    connectionManager.updateConfiguration(languageServerSettings);
+    connectionManager.updateServerConfigurations(languageServerSettings);
+    connectionManager.updateLogging(
+      options.logAllCommunication,
+      options.setTrace
+    );
+  };
+  languageServerManager.sessionsChanged.connect(() => {
+    settingRegistry.transform(plugin.id, {
```

Comment:
I need this pattern because the transform callback will read the language server session list,  and we don't have control over the availability of this list. 
> If that pattern is needed, you should store the disposable object output by settingRegistry.transform and dispose the previous one before adding a new transform callback.

Applied